### PR TITLE
Add masked_fill to ATen XLA tensor

### PR DESCRIPTION
**Pull request by asuhan**
I had to add a Python test to exercise the 0-dimensional tensor fill value, there's no good way to do that from C++.

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -1384,6 +1385,25 @@ XLATensor XLATensor::unsqueeze(const XLATensor& input, xla::int64 dim) {
       input.GetDevice());
 }
 
+XLATensor XLATensor::masked_fill(const XLATensor& input, const XLATensor& mask,
+                                 const at::Scalar& value) {
+  // Expand mask to be the same size as input.
+  ir::NodePtr expanded_mask = ir::MakeNode<ir::ops::Expand>(
+      mask.GetIrValue(), XlaHelpers::ShapeSizes(input.shape()));
+  return Create(ir::MakeNode<ir::ops::MaskedFill>(input.GetIrValue(),
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.cpp]: input.CreateFrom()

File: torch_xla/csrc/ops/masked_fill.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  xla::XlaOp input = loctx->GetOutputOp(operand(0));
+  xla::XlaOp mask = xla::Ne(
+      loctx->GetOutputOp(operand(1)),
+      XlaHelpers::ScalarValue(0, xla::PrimitiveType::U8, loctx->builder()));
+  // Input shape is the same as output shape.
+  const xla::Shape& input_shape = shape();
+  xla::PrimitiveType value_type = input_shape.element_type();
+  xla::XlaOp value = xla::Broadcast(
+      XlaHelpers::ScalarValue(value_, value_type, input.builder()),
+      XlaHelpers::ShapeSizes(input_shape));
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/masked_fill.cpp]: shape.dimensions()

File: torch_xla/csrc/ops/masked_fill.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      value_(std::move(value)) {}
+
+XlaOpVector MaskedFill::Lower(LoweringContext* loctx) const {
+  xla::XlaOp input = loctx->GetOutputOp(operand(0));
+  xla::XlaOp mask = xla::Ne(
+      loctx->GetOutputOp(operand(1)),
+      XlaHelpers::ScalarValue(0, xla::PrimitiveType::U8, loctx->builder()));
+  // Input shape is the same as output shape.
+  const xla::Shape& input_shape = shape();
+  xla::PrimitiveType value_type = input_shape.element_type();
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/masked_fill.cpp]: No need for this local.

File: torch_xla/csrc/ops/masked_fill.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+MaskedFill::MaskedFill(const Value& input, const Value& mask, at::Scalar value)
+    : Node(OpKind(at::aten::masked_fill), {input, mask}, input.shape(),
+           /*num_outputs=*/1, ScalarHash(value)),
+      value_(std::move(value)) {}
+
+XlaOpVector MaskedFill::Lower(LoweringContext* loctx) const {
+  xla::XlaOp input = loctx->GetOutputOp(operand(0));
+  xla::XlaOp mask = xla::Ne(
+      loctx->GetOutputOp(operand(1)),
+      XlaHelpers::ScalarValue(0, xla::PrimitiveType::U8, loctx->builder()));
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/masked_fill.cpp]: Use operand(1)'s element_type()

File: torch_xla/csrc/ops/masked_fill.cpp (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/ops/masked_fill.cpp]: Done.

File: torch_xla/csrc/ops/masked_fill.cpp (hunk 4 above)
Comment (asuhan) [reply torch_xla/csrc/ops/masked_fill.cpp]: Done.

File: torch_xla/csrc/ops/masked_fill.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/ops/masked_fill.cpp]: Done.

File: torch_xla/csrc/tensor.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/tensor.cpp]: Done.

**Pull request merged**